core/domain/feedback_jobs_one_off.py

```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for feedback."""
+
+import ast
+
+from core import jobs
+from core.domain import feedback_services
+from core.platform import models
+
+(feedback_models,) = models.Registry.import_models([models.NAMES.feedback])
+
+
+class FeedbackThreadMessagesCountOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for calculating the distribution of username lengths."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.FeedbackMessageModel]
+
+    @staticmethod
+    def map(item):
+        yield (item.thread_id, 1)
```

Comment:
I'm not 100% sure we haven't deleted any messages. It might be safer to yield the message ID and pick the number that's one more than the max message ID to assign as the message count. In reduce(), also yield an error message (with the exp id and thread id and thread subject) if this number does not equal the number of messages in the list -- we can then run this when testing to verify.